lib/puppet/pops/loader/ruby_legacy_function_instantiator.rb

```diff
@@ -60,4 +67,41 @@ def self.get_binding(loader_injected_arg)
     binding
   end
   private_class_method :get_binding
+
+  def self.assert_code(code_string, result)
+    ripped = Ripper.sexp(code_string)
+    return if ripped.nil?  # Let the next real parse crash and tell where and what is wrong
+    ripped.each {|x| walk(x, result) }
+  end
+  private_class_method :assert_code
+
+  def self.walk(x, result)
+    return unless x.is_a?(Array)
+    first = x[0]
+    case first
+    when :fcall, :call
+      # Ripper returns a :fcall for a function call in a module (want to know there is a call to newfunction()).
+      # And it returns :call for a qualified named call
+      identity_part = find_identity(x)
+      result << :found_newfunction if identity_part.is_a?(Array) && identity_part[1] == 'newfunction'
+    when :def, :defs
+      # There should not be any calls to def in a 3x function
+      # Ripper returns an array [:def, ...] for a regular def name, and a [:defs ...] for a def self.name
+      identity_part = find_identity(x)
+      # assume there is nothing fancy and that there is always an array with name/line to use (or get nothing)
+      mname, mline = (identity_part.is_a?(Array) ? [identity_part[1], identity_part[2][1]] : [nil, nil]).map {|v| v.nil? ? '<unknown>' : v }
```

Comment:
Is `identity_part[2]` guaranteed to be an Array?

Reply:
As far as I know, yes. There is no documentation.

Reply:
Since it's not specified, I'd feel better checking that it is an Array. I could see our assumption breaking when ruby 2.6 comes out.